listener/src/listener.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright (c) 2019, ROSIN-project
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice, this
+list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+this list of conditions and the following disclaimer in the documentation
+and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+ * This tutorial demonstrates simple receiving of messages over the ROS system 
+ * using the rxros interface
+ */
+
+#include <rxros/rxros.h>
+#include "std_msgs/String.h"
+
+int main(int argc, char **argv)
+{
+  rxros::init(argc, argv, "listener");
+
+  rxros::observable::from_topic<std_msgs::String>("/chatter", 1000)
+    .subscribe ([] (const std_msgs::String& msg)
+        { ROS_INFO("I heard: [%s]", msg.data.c_str()); });
```

Comment:
There is also `ROS_INFO_STREAM(..)` which uses `std::ostream` and supports `<<` operators if that would be more appropriate here.

Reply:
Suggestion: make the lambda actually indent, instead of on a single/two line(s).

Reply:
Re: ROS_INFO, I switched it is more C++ idiomatic, but not much more (and it departs further from the original tutorial; the harm & gain are both minor)
Re: indentation: I don't know of any coding convention or rule for that. I tried to split lines differently but I don't like any other option; I don't want to put the lambda on one line, because the line becomes too long). I will just indent it more.